pkg/sql/set_schema.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package sql
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
+	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgcode"
+	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgerror"
+	"github.com/cockroachdb/cockroach/pkg/sql/privilege"
+	"github.com/cockroachdb/cockroach/pkg/sql/sqlbase"
+)
+
+// PrepareSetSchema verifies that a table/type can be set to the desired
+// schema and returns the schema id of the desired schema.
+func PrepareSetSchema(
+	params runParams, desc catalog.MutableDescriptor, schema string,
+) (sqlbase.ID, error) {
```

Comment:
I would do a check and error out at the top here if the desc isnt a table or type.